test/dynamo/test_activation_checkpointing.py

```diff
@@ -470,6 +470,7 @@ def fn(x):
         self._validate(fn, backend, x, skip_check=True)
 
     @torch._functorch.config.patch(recompute_views=True)
+    @torch._dynamo.config.patch(skip_nnmodule_hook_guards=False)
```

Comment:
@yf225 I am not sure if you wrote the test with Dynamo identifying that a hook has been applied, and therefore it needs to recompile. You introduced this in https://github.com/pytorch/pytorch/pull/135727